src/collectors.cc

```diff
@@ -79,8 +79,7 @@ static Obj TYPE_KERNEL_OBJECT;
 **    push a generator <gen>  with exponent <exp> onto the stack.
 **
 **  SC_PUSH_WORD( word, exp )
-**    push <word>  with global exponent <exp>  into the stack, the macro uses
-**    <word> and <exp> only once.
+**    push <word>  with global exponent <exp>  into the stack.
```

Comment:
I've adjusted this comment as we now evaluated `word` twice; and my guess is that part of the reason for that nasty hack here was to ensure we evaluate it only once.
After inspecting all callers to this macro, it should be perfectly safe to evaluate `word` twice.

Reply:
Maybe add an explicit comment that you do evaluate it twice? Is there some reason not to make this an inline function?

Reply:
I thought about changing it into an inline function, but decided against it (for now), for two reasons: (a) I wanted to keep the changes in this commit as small as possible; (b) implementation would actually be somewhat nasty, since we modify nw, lw, pw, ew, ge ; so we'd have to pass pointers to these in (so pointers to pointers to pointers to UInt...), and then replace `*++nw` by `*++*nw`, and I am sure that I'd win some kind of price for that (but not a good one)....

Well, I just realize this is in a C++ file, so we could also use a C++ reference, which would at least cut down on the number of `*` we have to use. But then people editing this would need to know about C++ references. Which aren't terribly complicated to learn about, but we agreed to be careful about using C++ features, and that we didn't want to have more and more of them creep in silently over time, so I am reluctant to do that...